stan/math/prim/fun/generalized_inverse.hpp

```diff
@@ -34,8 +35,11 @@ generalized_inverse(const EigMat& G) {
   if (G.size() == 0)
     return {};
 
-  if (G.rows() == G.cols())
-    return inverse(G);
+  if (G.rows() == G.cols()) {
```

Comment:
I don't know why `ldlt` gives back success. However, if I do a Cholesky instead it works. This will be much faster 

```c++
 if (G.rows() == G.cols()) {
     LLT<MatrixXd> llt_G = G.llt(); 
    if(llt_G.info() == Success)
      return G.inverse();
  }
```